add CAL_STATIC_CAPABILITY_NO_SEND_IMIP calendar capability

From: Ondrej Jirman <ondrej.jirman@zonio.net>

calendar/ChangeLog:

2008-04-06  Ondrej Jirman  <megous@megous.com>

	** Fix for bug #525233

	* gui/dialogs/comp-editor.c:
	* gui/itip-utils.c:
	Add new capability CAL_STATIC_CAPABILITY_NO_SEND_IMIP to disable sending
	of iMIPs and conversion of attachment URLs to 'CID:...' by evolution.
	This is used by calendar backends that want to send iTIPs and upload
	attachments to a calendar server.
---

 calendar/gui/dialogs/comp-editor.c |    2 +-
 calendar/gui/itip-utils.c          |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)


diff --git a/calendar/gui/dialogs/comp-editor.c b/calendar/gui/dialogs/comp-editor.c
index a5b2232..399d5e5 100644
--- a/calendar/gui/dialogs/comp-editor.c
+++ b/calendar/gui/dialogs/comp-editor.c
@@ -2613,7 +2613,7 @@ real_send_comp (CompEditor *editor, ECalComponentItipMethod method)
 			set_attendees_for_delegation (send_comp, address, method);
 	}
 
-	if (!e_cal_component_has_attachments (priv->comp)) {
+	if (!e_cal_component_has_attachments (priv->comp) || e_cal_get_static_capability (priv->client, CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 		if (itip_send_comp (method, send_comp, priv->client,
 					NULL, NULL, users)) {
 			g_object_unref (send_comp);
diff --git a/calendar/gui/itip-utils.c b/calendar/gui/itip-utils.c
index 9398bcf..e164e4e 100644
--- a/calendar/gui/itip-utils.c
+++ b/calendar/gui/itip-utils.c
@@ -1194,7 +1194,7 @@ itip_send_comp (ECalComponentItipMethod method, ECalComponent *send_comp,
 	/* Recipients */
 	destinations = comp_to_list (method, comp, users, FALSE);
 	if (method != E_CAL_COMPONENT_METHOD_PUBLISH) {
-		if (destinations == NULL) {
+		if (destinations == NULL || e_cal_get_static_capability (client, CAL_STATIC_CAPABILITY_NO_SEND_IMIP)) {
 			/* We sent them all via the server */
 			retval = TRUE;
 			goto cleanup;
